be/src/olap/rowset/segment_v2/inverted_index_writer.cpp

```diff
@@ -215,13 +221,9 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
     }
 
     void new_fulltext_field(const char* field_value_data, size_t field_value_size) {
-        if (_parser_type == InvertedIndexParserType::PARSER_ENGLISH) {
+        if (_parser_type == InvertedIndexParserType::PARSER_ENGLISH ||
+            _parser_type == InvertedIndexParserType::PARSER_CHINESE) {
             new_char_token_stream(field_value_data, field_value_size, _field);
-        } else if (_parser_type == InvertedIndexParserType::PARSER_CHINESE) {
```

Comment:
Is there any risk to merge english and chinese parser?

Reply:
use char to avoid copy data in both english and chinese analyzer